### ISSUE TITLE
Run the script using the windows version of IDA Pro

### ISSUE THREAD

**Issue by Colacalc** [Asteria-BCSD/Asteria#10]
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/209e6e96-f53a-4afc-a484-9a27fc85b14a)
I have been able to run ast_Generating.py with F5 decompilation to generate *.sqlite files; However, the script is currently unusable, mainly because it is difficult to debug.
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/7a1e9d18-d628-4886-818b-748ee2850ac3)
The data in the figure is generated after the ARM-O0-acl-2.2.53-chacl script is used。
It is not clear whether such figures are correct; In addition, it can be seen that the data processing of dataheper.py file can be obtained in.pth format. However, if the batch processing script cannot be used, it will be difficult to obtain this data. Thanks a million

**Comment by Colacalc** [comment]
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/f8341593-a8df-43df-ac19-1ad05cb88cff)
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/0448141c-0d49-441f-9cbe-98836aad208b)
The first graph is my feedback from the read of the second graph
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/c673d44a-b887-4985-9292-6f0debc0d6bd)
When the script is executed in the figure above, the problem with Figure 1 occurs. The command is as follows：cmd_list: ['', 'idat.exe', '-c', '-A', '-S"G:\\IDA_Pro_v7.0_Portable\\python\\ASTExtraction\\ast_generator.py -o O0 -g gcc -d F:\\buildroot-elf-5arch\\X86\\O0\\x86O0.sqlite"', 'F:\\buildroot-elf-5arch\\X86\\O0\\mydata.sqlite']

**Comment by Asteria-BCSD** [comment]
> 我按照第二张图的read进行执行

The 'cmd_list' variable printed in the log (in your first figure) indicates that your target binary file is located at *F:\buildroot-elf-5arch\X86\O0\mydata.sqlite*. However, it seems that this file is not suitable for IDA to handle since it is a database file. Please make sure to provide a binary file to IDA for analysis.

**Comment by Colacalc** [comment]
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/037db913-26c6-48f4-928e-fd41ec98803b)
Yes, thank you for your reply. The debug process found the problem, the database file path has been changed, but still did not execute, so the input is assumed to be x86\O0 data path and sqlite file save path. cmd: exe -c -A -S"G:\IDA_Pro_v7.0_Portable\python\ASTExtraction\ast_generator.py -o O0 -g gcc -d F:\buildroot-elf-5arch\X86\x86O0.sqlite" F:\buildroot-elf-5arch\X86\O0\acl-2.2.53\chacl. At present, it is found that IDA Pro generates a series of *.id0*.id1 and other files after analyzing the files. However, get_arch calls the file command to determine the structure of the file. I commented it out here, so some problems are generated. At present, the debugging is continuing.
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/b27736a0-727e-4cb7-a5e2-45782c9fedfa)

**Comment by Colacalc** [comment]
In the last question I added a simple filename judgment to the code that for now it is possible to skip these files and process binaries in the data set normally.

**Comment by Colacalc** [comment]
If i want to add the desired final training data with a.pth suffix, can i merge it with x86-O0.sqlite, x86-O1 and so on, or like x86-sqlite, arm.sqlite? Data processing is relatively slow at the moment. Unable to see if the data fit properly with the final model training.

**Comment by Colacalc** [comment]
Ten hours of running on a consumer-grade laptop yielded about six megabytes of data, which seems like a slow analysis process. If possible, could you share the processed data set you trained on?

**Comment by shouguoyang** [comment]
> 

Yes, we are working on it. Once uploaded, we will post the download url.

**Comment by Colacalc** [comment]
When I in performing the above steps such as ARMO0 sqlite, ARMO1. Sqlite, X86O0. Sqlite, try to use datahelper. Py generated. PTH format of training data, encountered the following error
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/6dafdc16-695c-45ac-bb33-2a86eca3c7b5)
And errors from self.loads_ast_tree(t[-2])
![image](https://github.com/Asteria-BCSD/Asteria/assets/92523788/52d8e76c-1eaf-47fe-9834-24c02dd1bd7a)
I'm not sure what data format I'll need later, so I can't debug it here

**Comment by shouguoyang** [comment]
> e working on it. Once uploaded, we will post the download

Please download the training dataset from https://zenodo.org/record/7927518#.ZGR3pk_P0Q8.

**Comment by Colacalc** [comment]
It seems the website can't be accessed

**Comment by fallinshadow** [comment]
不好意思打扰您，想请问一下您在ida pro上运行ast_Generating.py时是否已经安装了torch等库？如果是的话请问您的是如何为python2.7安装torch库的呢？

**Comment by Colacalc** [comment]
您是具体什么需求，如果是使用插件，是需要使用模型进行特征编码的，需要安装torch，我使用的是Python3,直接根据报错修改代码即可



| |
***@***.***
|
|
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年05月18日 20:52 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [Asteria-BCSD/Asteria] Run the script using the windows version of IDA Pro (Issue #10) |

不好意思打扰您，想请问一下您在ida pro上运行ast_Generating.py时是否已经安装了torch等库？如果是的话请问您的是如何为python2.7安装torch库的呢？

**Comment by fallinshadow** [comment]
您是直接为ida pro配置了python3吗？请问是什么版本的ida呢？

**Comment by Colacalc** [comment]
7.0的，装在系统里的python3.8



| |
***@***.***
|
|
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年05月18日 21:12 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [Asteria-BCSD/Asteria] Run the script using the windows version of IDA Pro (Issue #10) |

您是直接为ida pro配置了python3吗？请问是什么版本的ida呢？

**Comment by fallinshadow** [comment]
好的。谢谢您！

**Comment by Colacalc** [comment]
Excuse me, the dataset link you provided seems to be inaccessible
